compiler/src/dotty/tools/dotc/core/Denotations.scala

```diff
@@ -1226,11 +1226,10 @@ object Denotations {
     def select(prefix: Denotation, selector: Name): Denotation = {
       val owner = prefix.disambiguate(_.info.isParameterless)
       def isPackageFromCoreLibMissing: Boolean =
-        owner.symbol == defn.RootClass &&
-        (
-          selector == nme.scala || // if the scala package is missing, the stdlib must be missing
-          selector == nme.scalaShadowing // if the scalaShadowing package is missing, the dotty library must be missing
-        )
+        // if the scala package is missing, the stdlib must be missing
+        (owner.symbol == defn.RootClass && selector == nme.scala) ||
+        // if the scala.compiletime package is missing, the dotty library must be missing
+        (prefix.symbol.owner == defn.RootClass && prefix.name == nme.scala && selector == nme.compiletime)
```

Comment:
This isn't really useful, the point of checking for scalaShadowing before is that we explicitly force it when starting the compiler, but the same isn't true for scala.compiletime, so there is no reason to special-case it here.

Reply:
Should we force something in the dotty library to be able to detect if it is not on the classpath. Or should we drop the check and just fail with a random non existent symbol error?

Reply:
I would drop the check for now, we want to avoid cascade of errors when symbols are missing, but this should be achievable without special casing dotty-library

Reply:
I dropped the check